src/server/http/xsrf.js

```diff
@@ -4,18 +4,49 @@ export default function (kbnServer, server, config) {
   const disabled = config.get('server.xsrf.disableProtection');
   const versionHeader = 'kbn-version';
   const xsrfHeader = 'kbn-xsrf';
+  const contentTypeHeader = 'content-type';
+  const allowedRequestMediaTypes = ['application/json', 'application/x-ndjson'];
 
   server.ext('onPostAuth', function (req, reply) {
     if (disabled) {
       return reply.continue();
     }
 
     const isSafeMethod = req.method === 'get' || req.method === 'head';
-    const hasVersionHeader = versionHeader in req.headers;
-    const hasXsrfHeader = xsrfHeader in req.headers;
+    if (!isSafeMethod) {
```

Comment:
I'll admit I'd prefer an early return *without* having to call reply.continue() as well, but the precedent has already been set with the `disabled` check at the beginning of the function.  There are actually already 2 `reply.continue` calls in the function, so this would make 3.